addons/point_of_sale/static/src/js/screens.js

```diff
@@ -2250,11 +2250,59 @@ define_action_button({
     },
 });
 
+
+/*--------------------------------------*\
+ |         THE LOGIN SCREEN           |
+\*======================================*/
+
+// The login screen enables employees to log in to the PoS
+// at startup or after it was locked, with either barcode, pin, or both.
+
+var LoginScreenWidget = ScreenWidget.extend({
+    template: 'LoginScreenWidget',
+
+    /**
+     * @override
+     */
+    show: function() {
+        var self = this;
+        this.$('.select-employee').click(function() {
+            self.gui.select_employee({
+                'security': true,
+                'current_employee': self.pos.get_cashier(),
+                'title':_t('Change Cashier'),})
+            .then(function(employee){
+                self.pos.set_cashier(employee);
+                $('.username').text(employee.name);
+                self.unlock_screen();
+            });
+        });
+        this._super();
+    },
+
+    /**
+     * @override
+     */
+    barcode_cashier_action: function(code) {
+        this._super(code);
+        this.unlock_screen();
+    },
+
+    unlock_screen: function() {
+        var screen = (this.gui.pos.get_order() ? this.gui.pos.get_order().get_screen_data('previous-screen') : this.gui.startup_screen) || this.gui.startup_screen;
+        this.gui.show_screen(screen);
```

Comment:
@lap-odoo This is more a functional question. When an employee logs in via the lock screen, should we always show:

- the startup screen of the POS (e.g. the tables in restaurant mode) or,
- the screen that was shown when the previous employee logged out (e.g. some in progress order)